api/pom.xml

```diff
@@ -51,11 +51,11 @@
       <artifactId>dom4j</artifactId>
       <scope>compile</scope>
     </dependency>
+
+    <!-- Needed for runtime testing -->
     <dependency>
       <groupId>org.springframework.security</groupId>
       <artifactId>spring-security-core</artifactId>
-      <version>${spring-security.version}</version>  <!-- see maven-parent-pom -->
-      <scope>compile</scope>
```

Comment:
If this artifact is just needed during tests (as mentioned in the comment https://github.com/pentaho/pentaho-platform/pull/4341/files#diff-8658bb96fff6ac848cd28d3242fce067R55 ) then scope should be test `<scope>test</scope>`.

Reply:
Nop, also needed for compilation...just checked

Reply:
If I put <scope>test</scope> it doesn't compile